File: dotnet/system/workspace/typescript/libs/workspace/domain/system/src/lib/data/equals.ts
import { PropertyType } from '@allors/workspace/meta/system';
import { CompositeTypes, UnitTypes } from "../Types";
import { ParameterizablePredicate } from "./ParameterizablePredicate";

export interface Equals extends ParameterizablePredicate {
  propertyType: PropertyType;
  value?: UnitTypes;
  object?: CompositeTypes;
}

File: dotnet/system/workspace/csharp/allors.workspace.domain/data/equals.cs
// <copyright file="Equals.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Data
{

    using Meta;

    public class Equals : IPropertyPredicate
    {
        public string[] Dependencies { get; set; }

        public Equals(IPropertyType propertyType = null) => this.PropertyType = propertyType;

        /// <inheritdoc/>
        public IPropertyType PropertyType { get; set; }

        public IObject Object { get; set; }

        public object Value { get; set; }

        public IRoleType Path { get; set; }

        public string Parameter { get; set; }

        public void Accept(IVisitor visitor) => visitor.VisitEquals(this);
    }
}
